src/main/java/com/internet/shop/db/Storage.java

```diff
@@ -0,0 +1,39 @@
+package com.internet.shop.db;
+
+import com.internet.shop.model.Order;
+import com.internet.shop.model.Product;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.model.User;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Storage {
+    public static final List<Product> products = new ArrayList<>();
+    public static final List<Order> orders = new ArrayList<>();
+    public static final List<ShoppingCart> shoppingCarts = new ArrayList<>();
+    public static final List<User> users = new ArrayList<>();
+    private static Long productId = 0L;
+    private static Long orderId = 0L;
+    private static Long shoppingCartId = 0L;
+    private static Long userId = 0L;
+
+    public static void addProduct(Product product) {
+        product.setId(productId++);
```

Comment:
in this case, you will have zero ID for your first element. it is not ok. Try to use the prefix form of this operator

Reply:
it's true. but why zero ID is bad?

Reply:
In a real Data Base, MySQL, for example, a row can not have zero id. So numeration will start from 1 there. In this case, I think it can be okay, but keep in mind this info 👍

Reply:
ok, thank you!

Reply:
zero id can be in database but i would agree that in most cases it's start from 1. In case with storage it's up to you